src/test/java/org/assertj/core/api/recursive/comparison/RecursiveComparisonAssert_isNotEqualTo_Test.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.api.recursive.comparison;
+
+import org.assertj.core.internal.objects.data.Person;
+import org.assertj.core.test.CartoonCharacter;
+import org.assertj.core.test.Jedi;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.assertj.core.api.recursive.comparison.Color.GREEN;
+import static org.assertj.core.test.NeverEqualComparator.NEVER_EQUALS_STRING;
+
+@DisplayName("RecursiveComparisonAssert isNotEqualTo")
+public class RecursiveComparisonAssert_isNotEqualTo_Test {
+  @Test
+  public void should_pass_when_either_actual_or_expected_is_null() {
+    // GIVEN
+    Person actual = null;
+    Person other = new Person();
+    // THEN
+    assertThat(actual).usingRecursiveComparison()
+                      .isNotEqualTo(other);
+    assertThat(other).usingRecursiveComparison()
+                     .isNotEqualTo(actual);
+
+  }
+
+  @Test
+  public void should_fail_when_actual_and_expected_are_null() {
+    // GIVEN
+    Person actual = null;
+    Person other = null;
+    // THEN
+    assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> assertThat(actual).usingRecursiveComparison()
+                                                                                       .isNotEqualTo(other));
+  }
```

Comment:
Use `AssertionsUtil.expectAssertionError` and verify that the proper message is built with `shouldNotBeEqualComparingFieldByFieldRecursively`, for that look at `RecursiveComparisonAssert_isEqualTo_BaseTest` for the base code to do that.

Reply:
> Use `AssertionsUtil.expectAssertionError` and verify that the proper message is built with `shouldNotBeEqualComparingFieldByFieldRecursively`, for that look at `RecursiveComparisonAssert_isEqualTo_BaseTest` for the base code to do that.

Should I create one base test class for isNotEqualTo like RecursiveComparisonAssert_isEqualTo_BaseTest?
I think almost all of the features have been tested before in RecursiveComparisonAssert_isEqualTo_xxx_Test. So do I need only one class to write all the tests on isNotEqualTo?